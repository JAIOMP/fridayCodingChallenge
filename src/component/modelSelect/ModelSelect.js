import React, { useEffect, useContext, useState } from 'react';
import SelectItem from '../common/selectItem/SelectItem'
import VehicleContext from "../../provider/VehicleDataContext";
import './ModelSelect.css';

const ModelSelect = () => {
    const [models, setModels] = useState([]);
    const { selectedBrand, setSelectedModel, setError } = useContext(VehicleContext);

    useEffect(() => {
        setError('');
        fetch(`http://localhost:8080/api/models?make=${selectedBrand}`)
            .then(res => res.json())
            .then(data => {
                if(data.length === 0) {
                    setError('EMPTY_MODEL_DATA');
                }
                setModels(data)
            }).catch(() => {
                setError('API_ERROR');
        });
    }, [selectedBrand]);

    const onSelect = (event) => {
        setSelectedModel(event.target.value);
    };

    return (
        <>
            {models.length > 0  &&
            <SelectItem data={models} dataFor="model" onSelect={onSelect}/>}
        </>
    );
};

export default ModelSelect;
