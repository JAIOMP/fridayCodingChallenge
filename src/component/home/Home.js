import React, {useContext} from 'react';
import BrandSelect from "../brandSelect/BrandSelect";
import ModelSelect from "../modelSelect/ModelSelect";
import VehicleDetails from "../vehicleDetails/VehicleDetails";
import VehicleContext from "../../provider/VehicleDataContext";
import {ERRORS} from "../constant";

const Home = () => {
    const { selectedBrand, selectedModel, error } = useContext(VehicleContext);
    return (
        <div>
            <BrandSelect/>
            {selectedBrand && <ModelSelect/>}
            {selectedModel && <VehicleDetails/>}
            {
                error !== '' &&
                <div className="error-message">
                    {ERRORS[error]}
                </div>
            }
        </div>
    );
};

export default Home;
