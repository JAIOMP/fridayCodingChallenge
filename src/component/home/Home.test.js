import React from 'react';
import {render} from '@testing-library/react';
import Home from './Home';
import VehicleContext from "../../provider/VehicleDataContext";
import ModelSelect from "../modelSelect/ModelSelect";

jest.mock("../brandSelect/BrandSelect", () => () => <div>Brand Select</div>);
jest.mock("../modelSelect/ModelSelect", () => () => <div>Model Select</div>);
jest.mock("../vehicleDetails/VehicleDetails", () => () => <div>Vehicle Details</div>);
describe('Home', () => {
    const contextData = {
        selectedModel: "Edge",
        selectedBrand: "Ford",
        error: "",
        setError: jest.fn(),
    };

    it('should render Home', async () => {
        const {container} = render(
            <VehicleContext.Provider value={contextData}>
                <Home />
            </VehicleContext.Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
