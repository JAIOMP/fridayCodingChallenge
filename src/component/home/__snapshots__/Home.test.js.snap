// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render Home 1`] = `
<div>
  <div>
    <div>
      Brand Select
    </div>
    <div>
      Model Select
    </div>
    <div>
      Vehicle Details
    </div>
  </div>
</div>
`;
